botocore/configprovider.py

```diff
@@ -546,24 +553,26 @@ def resolve_auto_mode(self, region_name):
         return 'standard'
 
     def _update_provider(self, config_store, variable, value):
-        provider = config_store.get_config_provider(variable)
+        provider = copy.copy(config_store.get_config_provider(variable))
         default_provider = ConstantProvider(value)
-        if isinstance(provider, ChainProvider):
+        if provider is None:
+            provider = default_provider
+        elif isinstance(provider, ChainProvider):
             provider.set_default_provider(default_provider)
-            return
         elif isinstance(provider, BaseProvider):
-            default_provider = ChainProvider(
-                providers=[provider, default_provider]
-            )
-        config_store.set_config_provider(variable, default_provider)
+            provider = ChainProvider(providers=[provider, default_provider])
+        config_store.set_config_provider(variable, provider)
```

Comment:
We should make sure to add unit tests for the smart default factory that we do not mutate the chain providers nor the section providers (nor their sub-chain providers) as well.